crates/rust-analyzer/src/main_loop/handlers.rs

```diff
@@ -815,6 +808,22 @@ pub fn handle_code_action(
         }
     }
 
+    // If the client only supports commands then filter the list
+    // and remove and actions that depend on edits.
+    if !world.config.client_caps.code_action_literals {
+        res = res
```

Comment:
I think `.retain` would be a more convenietn API to use here

Reply:
ah no, we do need to transform things... https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.drain_filter would work, but it's nighly

Reply:
I added a comment to move to `drain_filter` once stable.